src/comm.jl

```diff
@@ -1,264 +1,1531 @@
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+1] = req
+            emsg += 1
+         end
+
+         if rowMin && colMax
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm
+            srcOffset = dx - 1
+            for field in fields
+               for i in 0:(dz-1)
+                  domain.commDataSend[offset+i] = field[srcOffset+i*dx*dy]
+               end
+               offset += dz
+            end
+            idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + 1
+            src = view(domain.commDataSend, idx:(idx+(xferFields * dz)))
+            otherRank = myRank - domain.tp + 1
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+1] = req
+            emsg += 1
+         end
+
+         if rowMax && planeMin
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm
+            srcOffset = dx*(dy - 1)
+            for field in fields
+               for i in 0:(dx-1)
+                  domain.commDataSend[offset+i] = field[srcOffset+i]
+               end
+               offset += dx
+            end
+            idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + 1
+            src = view(domain.commDataSend, idx:(idx+(xferFields * dx)))
+            otherRank = myRank + domain.tp^2 + domain.tp
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+1] = req
+            emsg += 1
+         end
+
+         if colMax && planeMin
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm
+            srcOffset = dx - 1
+            for field in fields
+               for i in 0:(dy-1)
+                  domain.commDataSend[offset+i] = field[srcOffset+i*dx]
+               end
+               offset += dy
+            end
+            idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + 1
+            src = view(domain.commDataSend, idx:(idx+(xferFields * dy)))
+            otherRank = myRank - domain.tp^2 + 1
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+1] = req
+            emsg += 1
+         end
+
+         if rowMin && colMin && planeMin
+            # corner at domain logical coord (0, 0, 0) */
+            offset = pmsg * maxPlaneComm + emsg * maxEdgeComm +  cmsg * CACHE_COHERENCE_PAD_REAL
+            for (fi, field) in enumerate(fields)
+               domain.commDataSend[offset+fi] = field[1]
+            end
+            idx = pmsg * maxPlaneComm + emsg * maxEdgeComm + cmsg * CACHE_COHERENCE_PAD_REAL + 1
+            src = view(domain.commDataSend, idx:(idx+(xferFields * 1)))
+            otherRank = myRank - domain.tp^2 - domain.tp - 1
+            req = MPI.Isend(src, otherRank, msgType + myRank, domain.comm)
+            domain.sendRequest[pmsg+emsg+cmsg+1] = req
+            cmsg += 1
+         end
+
+
```

Comment:
Obviously missing from https://github.com/wsmoses/Enzyme-MPI/blob/d211d3522628628cc3990a2470639b3159104a37/lulesh-comm.cc#L767 to end

Reply:
I seemingly forgot to push